sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -338,26 +318,39 @@ def _buildprovider(
     else:
         user_agent = USER_AGENT
 
-    if connection_string:
-        provider._client = AzureAppConfigurationClient.from_connection_string(
-            connection_string,
-            user_agent=user_agent,
-            retry_total=retry_total,
-            retry_backoff_max=retry_backoff_max,
-            **kwargs
+    clients = []
+    if connection_string and endpoint:
+        clients.append(
+            ConfigurationClientWrapper.from_connection_string(
+                endpoint, connection_string, user_agent, retry_total, retry_backoff_max, **kwargs
+            )
         )
+        failover_endpoints = find_auto_failover_endpoints(endpoint)
```

Comment:
This is a synchronous call, right? The start-up has to wait it to return here, it depends on how long the timeout period for srv lookup, we'd better ensure the timeout period is not too long or make it to be an asynchronous call.

Reply:
@RichardChen820 we have a separate async library if a customer is looking for async. This PR just has the sync change, I'll have a separate PR up once we have this figured out it is mostly the same, minus the async part.

Looking at the code each request has a 2 second timeout value. Should that be good enough?

Reply:
Each request timeout in 2 seconds is good to me.

Reply:
The request for DNS resolution takes 2 seconds. But we also have 5 second retry. So each request can take upto 7 seconds to find auto failover endpoints (which also sounds like an acceptable delay for startup).